src/libs/Navigation/AppNavigator/createCustomStackNavigator/CustomRouter.ts

```diff
@@ -1,36 +1,31 @@
-import {StackRouter} from '@react-navigation/native';
-import lodashFindLast from 'lodash/findLast';
-import _ from 'underscore';
+import {NavigationState, PartialState, RouterConfigOptions, StackNavigationState, StackRouter} from '@react-navigation/native';
+import {ParamListBase} from '@react-navigation/routers';
 import NAVIGATORS from '@src/NAVIGATORS';
 import SCREENS from '@src/SCREENS';
+import type {ResponsiveStackNavigatorRouterOptions} from './types';
 
-/**
- * @param {Object} state - react-navigation state
- * @returns {Boolean}
- */
-const isAtLeastOneCentralPaneNavigatorInState = (state) => _.find(state.routes, (r) => r.name === NAVIGATORS.CENTRAL_PANE_NAVIGATOR);
+type State = NavigationState | PartialState<NavigationState>;
 
-/**
- * @param {Object} state - react-navigation state
- * @returns {String}
- */
-const getTopMostReportIDFromRHP = (state) => {
+const isAtLeastOneCentralPaneNavigatorInState = (state: State): boolean => !!state.routes.find((route) => route.name === NAVIGATORS.CENTRAL_PANE_NAVIGATOR);
+
+const getTopMostReportIDFromRHP = (state: State): string => {
     if (!state) {
         return '';
     }
-    const topmostRightPane = lodashFindLast(state.routes, (route) => route.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR);
 
-    if (topmostRightPane) {
+    const topmostRightPane = state.routes.filter((route) => route.name === NAVIGATORS.RIGHT_MODAL_NAVIGATOR).at(-1);
+
+    if (topmostRightPane?.state) {
         return getTopMostReportIDFromRHP(topmostRightPane.state);
     }
 
-    const topmostRoute = lodashFindLast(state.routes);
+    const topmostRoute = state.routes.at(-1);
 
-    if (topmostRoute.state) {
+    if (topmostRoute?.state) {
         return getTopMostReportIDFromRHP(topmostRoute.state);
     }
 
-    if (topmostRoute.params && topmostRoute.params.reportID) {
+    if (topmostRoute?.params && 'reportID' in topmostRoute.params && typeof topmostRoute.params.reportID === 'string' && topmostRoute.params.reportID) {
```

Comment:
Curious why `&& typeof topmostRoute.params.reportID === 'string' && topmostRoute.params.reportID` is needed? Shouldn't `'reportID' in topmostRoute.params` be enough?

Reply:
@allroundexperts Not really, we need to narrow down the type to make sure `reportID` is `string`, that's why